community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/StartsWithAcceptanceTest.scala

```diff
@@ -135,107 +99,71 @@ class LikeAcceptanceTest extends ExecutionEngineFunSuite with QueryStatisticsTes
+  test("STARTS WITH against a null value returns no matches") {
+    val result = executeWithAllPlanners("MATCH (a) WHERE a.name STARTS WITH NULL RETURN a")
 
     result.toList shouldBe empty
   }
 
 
-  test("NOT LIKE against a null value returns no matches") {
-    val result = executeWithAllPlanners("MATCH (a) WHERE a.name NOT LIKE NULL RETURN a")
-
-    result.toList shouldBe empty
-  }
-
-  test("ILIKE against a null value returns no matches") {
-    val result = executeWithAllPlanners("MATCH (a) WHERE a.name ILIKE NULL RETURN a")
+  test("NOT STARTS WITH against a null value returns no matches") {
+    val result = executeWithAllPlanners("MATCH (a) WHERE NOT a.name STARTS WITH NULL RETURN a")
 
     result.toList shouldBe empty
   }
 
-  test("NOT ILIKE against a null value returns no matches") {
-    val result = executeWithAllPlanners("MATCH (a) WHERE a.name NOT ILIKE NULL RETURN a")
+  test("ENDS WITH against a null value returns no matches") {
+    val result = executeWithAllPlanners("MATCH (a) WHERE a.name ENDS WITH NULL RETURN a")
 
     result.toList shouldBe empty
   }
 
-  // *** TESTS OF _
 
-  test("_ should match exactly one character") {
-    val a1 = createNode(Map("prop" -> "value"))
-    createNode(Map("prop" -> "valu"))
+  test("NOT ENDS WITH against a null value returns no matches") {
+    val result = executeWithAllPlanners("MATCH (a) WHERE NOT a.name ENDS WITH NULL RETURN a")
 
-    val result = executeWithAllPlanners("MATCH a WHERE a.prop LIKE 'valu_' RETURN a")
-
-    result.toList should equal(Seq(Map("a" -> a1)))
-  }
-
-  test("one letter at the start of a string") {
-    val result = executeWithAllPlanners("MATCH (a) WHERE a.name LIKE '_BCDEF' RETURN a")
-
-    result.toList should equal(Seq(Map("a" -> aNode)))
-  }
-
-  test("one letter at the end of a string") {
-    val result = executeWithAllPlanners("MATCH (a) WHERE a.name LIKE 'ABCDE_' RETURN a")
-
-    result.toList should equal(Seq(Map("a" -> aNode)))
-  }
-
-  test("first and last letter are not known") {
-    val result = executeWithAllPlanners("MATCH (a) WHERE a.name LIKE '_bcde_' RETURN a")
-
-    result.toList should equal(Seq(Map("a" -> cNode)))
-  }
-
-  test("single letter in the middle of a string") {
-    val result = executeWithAllPlanners("MATCH (a) WHERE a.name LIKE 'AB_DEF' RETURN a")
-
-    result.toList should equal(Seq(Map("a" -> aNode)))
+    result.toList shouldBe empty
   }
 
-  test("underscore must match at least one letter") {
-    val result = executeWithAllPlanners("MATCH (a) WHERE a.name LIKE 'ABC_DEF' RETURN a")
+  test("CONTAINS against a null value returns no matches") {
+    val result = executeWithAllPlanners("MATCH (a) WHERE a.name STARTS WITH NULL RETURN a")
 
     result.toList shouldBe empty
   }
 
-  test("none of the things") {
-    val result = executeWithAllPlanners("MATCH (a) WHERE a.name LIKE '_' RETURN a")
+
+  test("NOT CONTAINS against a null value returns no matches") {
```

Comment:
At this point I was of the mistaken assumption that we would have 'NOT CONTAINS' as infix, like we had 'NOT LIKE'. But later realised that was not the case.